apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/lib/site-intent/use-site-intent.js

```diff
@@ -1,18 +1,28 @@
 import apiFetch from '@wordpress/api-fetch';
-import { useState, useEffect, useCallback } from '@wordpress/element';
+import { useQuery } from 'react-query';
 
 const useSiteIntent = () => {
-	const [ siteIntent, setSiteIntent ] = useState( '' );
-
-	const fetchSiteIntent = useCallback( () => {
-		apiFetch( { path: '/wpcom/v2/site-intent' } )
-			.then( ( result ) => setSiteIntent( result.site_intent ) )
-			.catch( () => setSiteIntent( '' ) );
-	}, [] );
-
-	useEffect( () => {
-		fetchSiteIntent();
-	}, [ fetchSiteIntent ] );
-	return siteIntent;
+	const queryInfo = useQuery(
+		'site-intent', // add Site ID ?
```

Comment:
Are we fetching in wp-admin context? That way store might already be limited to per-site and you wouldn't need the ID.

Otherwise, ID is a good idea.